hw/dv/tools/waves.tcl

```diff
@@ -0,0 +1,45 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+#
+# Generic TCL included by tool-specific scripts when wave dumping is
+# enabled.
+#
+# This is used by all supported simulators, and the driver scripts
+# (dvsim.py) need to make sure that we don't ask for an unsupported
+# dumping format (SHM with VCS, for example).
+#
+
+set dump_fmt none
+if {[info exists ::env(DUMP_FMT)]} {
+    set dump_fmt "$::env(DUMP_FMT)"
+} else {
+    puts "ERROR: No DUMP_FMT specified for wave dumping."
+    quit
+}
+
+set tb_top "tb"
+if {[info exists ::(TB_TOP)]} {
+    set tb_top "$::env(TB_TOP)"
+}
+
+if {"$dump_fmt" == "fsdb"} {
+    # The fsdbDumpvars +all command dumps everything: memories,
+    # MDA signals, structs, unions, power and packed structs.
+    puts "Dumping waves with VERDI to waves.fsdb"
+    fsdbDumpfile  "waves.fsdb"
+    fsdbDumpvars  0 $tb_top +all
+    fsdbDumpSVA   0 $tb_top
```

Comment:
Hi Rupert @rswarbrick , sorry I am super late to the party. 
When I run xcelium and want to dump waveform in `fsdb`. I got this error:
```
Dumping waves with VERDI to waves.fsdb
xmsim: *E,TCLERR: invalid command name "fsdbDumpfile".
    while executing
"fsdbDumpfile  "waves.fsdb""
```
I think the fix would be adding a `call` prefix to these three lines. 
I tried to fix it, but besides adding another environment variable, I am not sure if there is a better way to do it..

Reply:
Oh drat. That's why I originally split the wave dumping code per tool - I assumed that different tools might have different syntax for each format.

@sriyerg: What do you think we should do to get this working properly? Would you prefer a "default" waves.tcl which could be overridden for a specific tool, or would you prefer one per tool? I guess the decision depends on whether "most" tools look the same. I don't think I know about that - maybe you do?

Reply:
@cindychip are we supporting fsdb with xcelium?
I was under the impression that required a special plugin from novas.
my understanding was that xcelium only supports shm and vcd/evcd as standard?

Reply:
@rasmus-madsen We did support it before, and this PR broke it. See the (deleted) `hw/dv/tools/xcelium/xcelium_fsdb.tcl`.

Reply:
obviously - I excuse my dumb question